docs/adr/0038-use-component-library.md

```diff
@@ -0,0 +1,59 @@
+# Use a React component library
+
+- Status: proposed
+- Deciders: Reino Muhl
+- Date: 2023-04-03
+
+## Context and Problem Statement
+
+As part of a major refactor of the Subscription Platform, the team is trying to simplify and reduce the number of components used by the Subscription Platform web app. To that end the team is investigating whether or not to replace some of the more commonly used components, such as a Tooltip, DialogModal, etc., with a component library that provides out of the box features such as accessibility and screen reader support, for example.
+
+Additionally, maintaining and implementing the existing components sometimes comes with additional developer cost. For example, during a recent refactor introducing TailwindCSS, it was difficult to refactor items in the bespoke component library due to deeply nested styling.
```

Comment:
Can you maybe elaborate why you think option A wouldn't help mitigate future refactors? One of the big reasons Unstyled components stood out to me, is that there wasn't any common styling, and you could style the component the way you needed it where it was used.

For example, issues we ran into when refactoring Payments to use Tailwind, is that the `<Input>`/`<Button>` components, from the Payments component library, had some styles applied to it, which made it tricky to refactor. Using an unstyled library ensures that won't happen.

Reply:
If we went with option A and still provided our own custom/deeply nested CSS, like the old content-server SCSS, I don't think it would help a whole lot. If we're using something more easily updated like Tailwind, it would. I only meant to highlight that option A in itself may not help, but the switch to more maintainable CSS does.

Reply:
Ah I see. OK great, thank you for clarifying.